### ISSUE TITLE
feature: Allow merging/extending `select` arrays in environments

### ISSUE THREAD

**Issue by melgazar9** [meltano/meltano#8063]
### Feature scope

API

### Description

I would like meltano.yml to be able to select all with exception to one or two <tables> <collections>. For example, I have a tap-mongodb tap that I am running, which selects 80+ documents in the parent configuration and is set up to run for dev, testing, staging, and production with the same meltano.yml file only changing the credentials. The problem I'm facing is that one collection in testing environment has inconsistent datatypes and breaks the meltano pipeline. Since this is not expected and doesn't occur in staging & production I would like to remove this collection from select in meltano.yml rather than list out 79 collections I want to select.

select
  - !collectionX.*

**Comment by tayloramurphy** [comment]
@melgazar9 agreed this would be useful. I don't think it'd be _too_ hard to add, but we're not going to prioritize it at this time. cc @edgarrmondragon
